File: pset6/caesar.py
import sys

if len(sys.argv) != 2 or int(sys.argv[1]) <= 0:
    print('false')
    exit(1)
    
key = int(sys.argv[1])
text = input('plaintext: ')

if text:
    print('ciphertext: ', end='')
    for i in range(len(text)):
        if text[i].isalpha() and text[i].isupper():
            code_text = ((ord(text[i]) - ord('A') + key) % 26) + ord('A')
            print(chr(code_text), end='')
        elif text[i].isalpha() and text[i].islower():
            code_text = ((ord(text[i]) - ord('a') + key) % 26) + ord('a')
            print(chr(code_text), end='')
        else:
            print(text[i], end='')
    print()
            

File: pset2/caesar/caesar.c
/*  caesar.c
*   Programm by Serhii Sydorenko
*/

#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int main(int argc, string argv[])
{
    // check not NULL argument
    if ( argc != 2)
    {
        printf("fail\n");
        return 1;
    }
    
    // convert k to integer
    int key = atoi(argv[1]); 

    if (key <= 0)
    {
        printf("fail\n");
        return 1; 
    }
        
    printf("plaintext: ");
    string text = get_string();

    if (text != NULL)
    {
        printf("ciphertext: ");
        for (int i = 0, t = strlen(text); i < t; i++)
        {
            // Check the text of the letters and upper letters
            if (isalpha(text[i]) && isupper(text[i])) 
            {
                int code_text = ((text[i] - 'A' + key) % 26) + 'A';
                printf("%c", (char) code_text);
            
              // Check the text of the letters and lower letters  
            } else if (isalpha(text[i]) && islower(text[i])) 
            {
                int code_text = ((text[i] - 'a' + key) % 26) + 'a';
                printf("%c", (char) code_text);
            } else
            {
                int code_text = text[i];
                printf("%c", (char) code_text);
            }
        }
        printf("\n");
    }
    return 0;
}